internal/k3s/install.go

```diff
@@ -0,0 +1,221 @@
+package k3s
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/weka/gohomecli/internal/bundle"
+)
+
+const k3sImagesPath = "/var/lib/rancher/k3s/agent/images/"
+const defaultLocalStoragePath = "/opt/local-path-provisioner"
+
+var (
+	ErrExists = errors.New("k3s already installed")
+)
+
+var (
+	k3SInstallDir string
+	k3sBinary     string
+
+	k3sBundleRegexp = regexp.MustCompile(`k3s-(.*)\.(tar(\.gz)?)|(tgz)`)
+)
+
+func init() {
+	// by default install to current directory
+	exe, _ := os.Executable()
```

Comment:
All these manipulations connected with bundle files should be in internal/bundle